packages/sites/src/drafts/delete-draft.ts

```diff
@@ -0,0 +1,30 @@
+import { IModel, IHubRequestOptions } from "@esri/hub-common";
+import { removeItemResource } from "@esri/arcgis-rest-portal";
+import { _getDraftResourceName } from "./_get-draft-resource-name";
+
+/**
+ * Given an item id, removes the current draft resource if exists
+ * @param {*} siteOrPageId
+ * @param {*} hubRequestOptions
+ */
+export function deleteDraft(
+  siteOrPageModel: IModel,
+  hubRequestOptions: IHubRequestOptions
+) {
+  const {
+    item: { id, owner }
+  } = siteOrPageModel;
+  return _getDraftResourceName(id, hubRequestOptions).then(
+    (draftResourceName: string) => {
+      if (draftResourceName) {
+        return removeItemResource({
```

Comment:
you could consider `failSafe`-ing this call

Reply:
Tell me more. Why wouldn't we want it to throw if the draft didn't get removed?